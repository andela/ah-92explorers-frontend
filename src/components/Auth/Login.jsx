import React from 'react';
import { Link } from 'react-router-dom';
import '../../assets/auth/css/style.css';
import '../../assets/auth/css/bootstrap.min.css';
import '../../assets/auth/css/animate.css';
import Input from '../Common/Input';
import { Social } from './Social';


class LoginComponent extends React.Component {
  render() {
    const {
      loginError, onSubmit, onChange, email, password, errors,
    } = this.props;
    return (
      <div>
        { loginError && <p className="alert alert-danger">{loginError}</p>}
        <form id="loginForm" onSubmit={onSubmit} className="form mb-5">
          <Input
            name="email"
            label="Email"
            value={email}
            error={errors.email}
            onChange={onChange}
            placeholder="Email"
          />
          <Input
            name="password"
            label="Password"
            value={password}
            error={errors.password}
            onChange={onChange}
            type="password"
            placeholder="Password"
          />
          <div className="form-group mb-4">
            <span className="small">
              Donâ€™t have an account?
              {' '}
              <Link to="/signup">Join</Link>
            </span>
            <br />
            <span className="small">
              Forgot password?
              {' '}
              <Link to="/reset-password"> Click here</Link>
            </span>
          </div>
          <div className="form-group">
            <input type="submit" className="btn btn-primary btn-block" value="Log in with Email" />
          </div>
          <Social />
        </form>
      </div>
    );
  }
}

export default LoginComponent;
