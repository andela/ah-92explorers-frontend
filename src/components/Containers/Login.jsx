import React from 'react';
import { connect } from 'react-redux';
import { Redirect } from 'react-router-dom';
import validateInput from '../../middleware/validations/login';
import { login } from '../../redux/actions/actionCreators';
import LoginComponent from '../Auth/Login.jsx';

export class LoginContainer extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      email: '',
      password: '',
      errors: {},
    };
  }

  componentDidMount() {
    if (this.props.auth.isAuthenticated) {
      window.location.href = '/';
    }
  }

  onChange = (e) => {
    this.setState({ [e.target.name]: e.target.value });
  }

  onSubmit = (e) => {
    e.preventDefault();
    if (this.isValid()) {
      this.setState({ errors: {} });
      const { email, password } = this.state;
      this.props.login(email, password);
    }
  }

  isValid() {
    const { errors, isValid } = validateInput(this.state);
    if (!isValid) {
      this.setState({ errors });
    }
    return isValid;
  }

  render() {
    const {
      errors, email, password,
    } = this.state;
    const { isLoginSuccess, loginError } = this.props;

    if (isLoginSuccess) {
      return <Redirect to="/" />;
    }

    return (
      <div className="d-flex">
        <div className="form-wrap d-flex">
          <div className="align-self-center">
            <div className="form-wrap-inner align-self-center h-100">
              <a href="/" className="logo">
                <img src={require('../../assets/images/logo.png')} alt="" />
              </a>
              <h3 className="mb-4">Log in With Credentials</h3>
              <p className="mb-5">Foster inspiration and innovation by leveraging the modern web.</p>
              <LoginComponent
                errors={errors}
                email={email}
                password={password}
                loginError={loginError}
                onSubmit={this.onSubmit}
                onChange={this.onChange}
              />
            </div>
          </div>
        </div>
        <div className="img-wrap" style={{ backgroundImage: `url(${'https://cdn.dribbble.com/users/992274/screenshots/6206904/office_kit8-net.png'})` }} />
      </div>
    );
  }
}

const mapStateToProps = (state) => ({
  isLoginSuccess: state.login.isLoginSuccess,
  loginError: state.login.loginError,
  auth: state.login,
});

const mapDispatchToProps = (dispatch) => ({
  login: (email, password) => dispatch(login(email, password)),
});

export default connect(mapStateToProps, mapDispatchToProps)(LoginContainer);
