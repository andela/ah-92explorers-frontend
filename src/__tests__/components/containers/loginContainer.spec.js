import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import LoginComponent from '../../../components/Auth/Login.jsx';
import { LoginContainer } from '../../../components/Containers/Login.jsx';

const props = {
  isLoginSuccess: false,
  loginError: {},
  email: '',
  password: '',
  errors: {},
  login: jest.fn(),
  auth: { isAuthenticated: false },
  loading: false,
};

let wrapper;
const initialState = {};
// here it is possible to pass in any middleware if needed into //configureStore
const mockStore = configureStore();
let store;

describe('<LoginContainer />', () => {
  beforeEach(() => {
    // creates the store with any initial state or middleware needed
    store = mockStore(initialState);
    wrapper = shallow(<LoginContainer store={store} {...props} />);
  });

  it('should render Login container', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should check componentDidMount', () => {
    wrapper.instance().componentDidMount()
    wrapper.setState({email: 'ele@rlr.com'})
    expect(wrapper.instance().props.auth.isAuthenticated).toBeDefined();
    expect(jest.isMockFunction(window.location.href)).toBeDefined();
  });

  it('Should render <LoginComponent />  component', () => {
    expect(wrapper.find(LoginComponent)).toHaveLength(1);
  });

  it('Should give initial state', () => {
    expect(wrapper.state()).toBeDefined();
  });

  it('should handle on click login', () => {
    const fakeEvent = { preventDefault: () => {} };
    const component = wrapper.instance();
    component.onSubmit(fakeEvent);
    expect(component).toBeDefined();
  });

  it('should handle on change', () => {
    const component = wrapper.instance();
    const fakeEvent = { target: { name: 'title', value: 'THIS IS TITLE' } };
    component.onChange(fakeEvent);
    expect(component).toBeDefined();
  });
});
