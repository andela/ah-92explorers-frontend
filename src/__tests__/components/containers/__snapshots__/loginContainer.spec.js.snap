// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginContainer /> should render Login container 1`] = `
<div
  className="d-flex"
>
  <div
    className="form-wrap d-flex"
  >
    <div
      className="align-self-center"
    >
      <div
        className="form-wrap-inner align-self-center h-100"
      >
        <a
          className="logo"
          href="/"
        >
          <img
            alt=""
            src="test-file-sdub"
          />
        </a>
        <h3
          className="mb-4"
        >
          Log in With Credentials
        </h3>
        <p
          className="mb-5"
        >
          Foster inspiration and innovation by leveraging the modern web.
        </p>
        <LoginComponent
          email=""
          errors={Object {}}
          loginError={Object {}}
          onChange={[Function]}
          onSubmit={[Function]}
          password=""
        />
      </div>
    </div>
  </div>
  <div
    className="img-wrap"
    style={
      Object {
        "backgroundImage": "url(https://cdn.dribbble.com/users/992274/screenshots/6206904/office_kit8-net.png)",
      }
    }
  />
</div>
`;
